src/main/java/soot/jimple/internal/JInterfaceInvokeExpr.java

```diff
@@ -19,54 +19,65 @@
  */
 
 /*
- * Modified by the Sable Research Group and others 1997-1999.  
+ * Modified by the Sable Research Group and others 1997-1999.
  * See the 'credits' file distributed with Soot for the complete list of
  * contributors.  (Soot is distributed at http://www.sable.mcgill.ca/soot)
  */
 
-
-
-
-
-
 package soot.jimple.internal;
 
-import soot.*;
-import soot.jimple.*;
+import soot.SootClass;
+import soot.SootMethod;
+import soot.SootMethodRef;
+import soot.SootMethodRefImpl;
+import soot.Value;
+import soot.ValueBox;
+import soot.jimple.Jimple;
+
+import java.util.ArrayList;
+import java.util.List;
 
-import java.util.*;
+public class JInterfaceInvokeExpr extends AbstractInterfaceInvokeExpr {
 
-public class JInterfaceInvokeExpr extends AbstractInterfaceInvokeExpr
-{	
-    public JInterfaceInvokeExpr(Value base, SootMethodRef methodRef, List<? extends Value> args)
-    {
+    public JInterfaceInvokeExpr(Value base, SootMethodRef methodRef, List<? extends Value> args) {
         super(Jimple.v().newLocalBox(base), methodRef, new ValueBox[args.size()]);
 
         //Check that the method's class is resolved enough
         //CheckLevel returns without doing anything because we can be not 'done' resolving
         methodRef.declaringClass().checkLevelIgnoreResolving(SootClass.HIERARCHY);
+
         //now check if the class is valid
-        if(!methodRef.declaringClass().isInterface() && !methodRef.declaringClass().isPhantom()) {
-        	throw new RuntimeException("Trying to create interface invoke expression for non-interface type: " +
-        			methodRef.declaringClass() +
-        			" Use JVirtualInvokeExpr or JSpecialInvokeExpr instead!");
+        if (methodRef.isStatic()) {
```

Comment:
Here, you allow methods that aren't declared by interfaces but normal classes as well. Ist this correct? As I understand the language specification, this is illegal. Can you craft an example where an invokeinterface instruction can actually invoke a class declared method?